timesketch/lib/analyzers/feature_extraction.py

```diff
@@ -229,7 +266,46 @@ def extract_feature(self, name, config):
                 continue
 
             event_counter += 1
-            event.add_attributes({store_as: result[0]})
+            store_type = event.source.get(store_as)
+            if store_type and not overwrite_store_as:
+                continue
+            if isinstance(store_type, six.text_type):
+                store_type_list = False
+            elif isinstance(store_type, (list, tuple)):
+                store_type_list = True
+            if store_type_list:
```

Comment:
here you are starting to wade into the land of too much indentation and I would opt for you to create a new function

```
def _get_attribute_value(current_value, new_value, .....): 
....
```
(better to have this function at the module level, not at the class level, or have it at the class level and decorate it as a static method, since you don't need `self` and this is going to be potentially called quite a few times, so better to attempt to speed things up at the same time as you are making it more readable)

and then call it like this:

```
if store_type_list:
  attribute_value = self._get_attribute_value(current_value, result, .....
else:
  attribute_value = result[0]

event.add_attributes({store_as: attribute_value})
```

That way you are not mixing together the changes to the event and figuring out what the attribute it (separation of concerns) and you are not using up too much indentation that makes the code hard to read.

Reply:
you can also make it much easier in many ways by returning early if certain conditions are met.